### PR TITLE
How To Add Pure CSS Waves To Your Website

### LINKED ISSUE #4825: [Languages]  How to Add Pure CSS Wave Animations to Your Website
#### Proposed title of the article
How to Add Pure CSS Wave Animations to Your Website

#### Proposed article introduction
 Adding animations can take you to the next level when you need to make your website seem more lively and attractive to your users.  The downfall of adding animations to a website using Javascript, as that is the widely used language in adding dynamicity to a website, is animations lead to the website loading slower especially when the developer fails to select an optimized library. However, there is a turnaround to such a problem and that is using CSS and HTML only.
#### Key takeaways
1. The reader will get to know what CSS animations are, and a few of their advantages over Javascript animations.
2. A demonstration and detailed explanation of how to use the CSS animation property.
3. A detailed tutorial on how to make pure CSS wave animations on your website. 
#### Article quality
The article does not only provide a simple tutorial it also adds a detailed explanation of the difference between animations in CSS and Javascript and also introduces the reader to CSS animations.

### PR DESCRIPTION
**Pull request by lin-chez**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes

`This closes #4825

### CODE REVIEW

**Comment by jumamiller** [comment]
Add author details!

**Comment by lin-chez** [comment]
> Add author details!

Got it!

**Comment by lin-chez** [comment]
added the needed details. @jumamiller

**Comment by lin-chez** [comment]
What's wrong? @jumamiller

**Comment by jumamiller** [comment]
> What's wrong? @jumamiller

Kindly be a little patient. I have seen the updates!

**Comment by jumamiller** [comment]
> > What's wrong? @jumamiller
> 
> Kindly be a little patient. I have seen the updates!

I'll review your work as soon as possible!

**Comment by lin-chez** [comment]
thank you for the affirmation @jumamiller

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hi, kindly ensure all code snippets are explained!

**Comment by lin-chez** [comment]
Hallo. I believe they are. However, you can list the areas where you feel i
should expound on more. Thanks

**Review by lin-chez — COMMENTED** [review]
The changes highlighted at line 143 are an explanation of the code at line 146. I believe all code snippets are explained.

**Comment by hectorkambow** [comment]
@jumamiller  

If needed lets consider pinging our team for help - we have more than enough pull request. 

No need to assign yourself more than needed 👀

File: content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+### Introduction
+This article will show you how to add animations using CSS and HTML only. 
+
+### Prerequisites
+To follow along with this tutorial, you should have:
+
+- A basic knowledge of HTML and CSS.
+- Your favourite code editor.
+
+
```
Comment (jumamiller) [inline content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md:12]: Remove these extra spacings

File: content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+  0% {
+    margin-left: 0;
+  }
+  100% {
+    margin-left: -1600px;
+  }
+}
+```
+
+- We define a transform property with the value translate3d and set the y-value to go from a negative to a positive then back to a negative value to make the second child element move up and down, completing the building of a wavy animation.
```
Comment (jumamiller) [inline content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md:169]: You have typos on this paragraph. i'e `translate3d`.

File: content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+    transform: translate3d(0,-24px,0);
+  }
+  50% {
+    transform: translate3d(0,5px,0);
+  }
+}
+
+```
+
+Our final website will have a waves animation that moves from right to left.
```
Comment (jumamiller) [inline content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md:261]: Could we have screenshot of final output? or link to github?

File: content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
+Adding animations can take you to the next level by making your website seem livelier and more attractive to your users. The downfall of adding animations to a website using JavaScript, as this is the widely used language in adding dynamicity to a website, is animations lead to a website loading slower, especially when the developer fails to select an optimized library. However, there is a turnaround to such a problem, and that is using CSS and HTML only.
+
+### Introduction
+This article will show you how to add animations using CSS and HTML only.
```
Comment (jumamiller) [inline content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md:4]: This is too short?

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hi, work on the minor edits suggested!

File: content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md (hunk 4 above)
Comment (jumamiller) [reply content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md:4]: The introduction section!

**Comment by hectorkambow** [comment]
@jumamiller 
Thank you for the great feedback - lets take a look at the recent changes provided.

IF any other revisions are needed - lets provide some guidance on how to get the article up to par - and CLOSE the PR. To allow the author the proper time to make any and all needed corrections.

This is a duplicate PR - if any more changes were needed (in general) we should close the PR immediately especially if its a duplicate PR (generally there are always exceptions).

File: content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md (hunk 2 above)
Comment (lin-chez) [reply content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md:169]: that is not a typo it is a property value.

**Review by lin-chez — COMMENTED** [review]
changes completed.

**Comment by hectorkambow** [comment]
@jumamiller 
If any more major revisions are needed - we will be closing this PR asap. 👀

@lin-chez 
Please be sure you are going over the entire article for any other mistakes or errors, as our peer reviewers are not responsible to point out every mistake - that is the authors responsibility to properly review and proofread their work.

**Comment by lin-chez** [comment]
> Hi, work on the minor edits suggested!

@jumamiller Hallo! I believe all requested changes were met, kindly approve them or point out other sections that need my attention. Thanks.

File: content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
+
+### Introduction
+Adding animations can take you to the next level by making your website seem livelier and more attractive to your users. developers use animations to develop websites that attract more users, increase a websites presence and improve conversions. The downfall of adding animations to a website using JavaScript, as this is the widely used language in adding dynamicity to a website, is animations reduce the performance of a website, especially when the developer fails to select an optimized library. However, there is a turnaround to such a problem, and that is using CSS and HTML only. 
+
+### Goal
+In this article, we will add pure CSS and HTML animations to a website as shown in the video below:
+<iframe width="478" height="269" src="https://www.youtube.com/embed/sAjpeUh3bIs" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (jumamiller) [inline content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md:7]: Insert this video properly! You may check makrdown docs.

File: content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md [hunk 6]
```diff
@@ -0,0 +1,254 @@
+
+### Introduction
+Adding animations can take you to the next level by making your website seem livelier and more attractive to your users. developers use animations to develop websites that attract more users, increase a websites presence and improve conversions. The downfall of adding animations to a website using JavaScript, as this is the widely used language in adding dynamicity to a website, is animations reduce the performance of a website, especially when the developer fails to select an optimized library. However, there is a turnaround to such a problem, and that is using CSS and HTML only.
```
Comment (jumamiller) [inline content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md:3]: This is too long. Breakdown it down into paragraphs. Additionally, introduction is only meant to help readers to get general overview of what to expect. Make it as simple as possible!

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hi, you  very minor edits that I think you overlooked. For instance punctuation marks and paragraphs.

Feel free to reach me slack for any support.^(@Miller).

**Review by lin-chez — COMMENTED** [review]
@jumamiller Minor edits have been made to the PR kindly approve them. Thanks.

File: content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md (hunk 5 above)
Comment (jumamiller) [reply content/articles/how-to-add-pure-css-wave-animations-to-your-website/index.md:8]: Hi, it''s impossible to embed YouTube videos .Just give readers a link as I had said to you on Slack!

**Comment by jumamiller** [comment]
@lin-chez, I'm testing your codes currently. Could you kindly add a github repo for the complete site and provide a link?

**Comment by lin-chez** [comment]
> @lin-chez, I'm testing your codes currently. Could you kindly add a GitHub repo for the complete site and provide a link?

@jumamiller kindly clarify the need for a GitHub repo when the complete code is featured on the last segments of the article.

**Comment by jumamiller** [comment]
> > @lin-chez, I'm testing your codes currently. Could you kindly add a GitHub repo for the complete site and provide a link?
> 
> @jumamiller kindly clarify the need for a GitHub repo when the complete code is featured on the last segments of the article.

Cloning a project is easier than copy pasting?

**Review by jumamiller — APPROVED** [review]
I had to fork the repo for this article to act as my reference in the future. I really loved the animation!

**Comment by jumamiller** [comment]
> I had to fork the repo for this article to act as my reference in the future. I really loved the animation!

Hope it helps other devs out there!

**Comment by lin-chez** [comment]
@jumamiller Thank you for the wonderful remarks. I also thank you for your patience, I gained some tips on how to better my services. For that am grateful.😊

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**